### PR TITLE
Update __init__.py

### LINKED ISSUE #532: Problem with using domain socket as bind address
**Describe your environment** 
The OT-wsgi library throws error if domain socket is used for the bind address.

**Steps to reproduce**
Here is a test program:

```
import web
from time import sleep
from opentelemetry import trace
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import (
    ConsoleSpanExporter,
    SimpleSpanProcessor,
)
from opentelemetry.instrumentation.wsgi import OpenTelemetryMiddleware
from cheroot import wsgi

trace.set_tracer_provider(TracerProvider())
trace.get_tracer_provider().add_span_processor(
    SimpleSpanProcessor(ConsoleSpanExporter())
)

#tracer = trace.get_tracer(__name__)

urls = (
    '/', 'index'
)
class index:
    def GET(self):
        return "Hello, world!"

if __name__ == "__main__":
  app = web.application(urls, globals())
  func = app.wsgifunc()

  func = OpenTelemetryMiddleware(func)

  server = wsgi.WSGIServer("/tmp/lr.sock", func, server_name="localhost")
  server.start()
```

invocation:
```
(base) kamalh-mbp:~ kamalh$ echo -ne 'GET / HTTP/1.1\r\nHost: test.com\r\n\r\n' | nc -U /tmp/lr.sock
HTTP/1.1 500 Internal Server Error
Content-Length: 0
Content-Type: text/plain
```

Error from the program
```
(base) kamalh-mbp:opentelemetry kamalh$ python3 wsgi-lr.py
Overriding of current TracerProvider is not allowed
ValueError("invalid literal for int() with base 10: ''")
Traceback (most recent call last):
  File "/Users/kamalh/miniconda3/lib/python3.7/site-packages/cheroot/server.py", line 1287, in communicate
    req.respond()
  File "/Users/kamalh/miniconda3/lib/python3.7/site-packages/cheroot/server.py", line 1077, in respond
    self.server.gateway(self).respond()
  File "/Users/kamalh/miniconda3/lib/python3.7/site-packages/cheroot/wsgi.py", line 140, in respond
    response = self.req.server.wsgi_app(self.env, self.start_response)
  File "/Users/kamalh/miniconda3/lib/python3.7/site-packages/opentelemetry/instrumentation/wsgi/__init__.py", line 229, in __call__
    attributes=collect_request_attributes(environ),
  File "/Users/kamalh/miniconda3/lib/python3.7/site-packages/opentelemetry/instrumentation/wsgi/__init__.py", line 122, in collect_request_attributes
    result.update({SpanAttributes.NET_HOST_PORT: int(host_port)})
ValueError: invalid literal for int() with base 10: ''
```

**What is the expected behavior?**
Expect to see the server returning normally as in TCP sockets.

**What is the actual behavior?**
Error message. Please see the paste above.

**Additional context**

### PR DESCRIPTION
**Pull request by kamalhussain**
Adding a check to see if the host_port is empty. Previously, this used to throw ValueError exception if host_port is empty.

# Description

Adding a defensive a check to see if host_port is empty to avoid the ValueError exception being raised.

Fixes #532 

## Type of change

Please delete options that are not relevant.
- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
- [ ] Test A

Test program:
```
import web
from time import sleep
from opentelemetry import trace
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import (
    ConsoleSpanExporter,
    SimpleSpanProcessor,
)
from opentelemetry.instrumentation.wsgi import OpenTelemetryMiddleware
from cheroot import wsgi

trace.set_tracer_provider(TracerProvider())
trace.get_tracer_provider().add_span_processor(
    SimpleSpanProcessor(ConsoleSpanExporter())
)

#tracer = trace.get_tracer(__name__)

urls = (
    '/', 'index'
)
class index:
    def GET(self):
        return "Hello, world!"

if __name__ == "__main__":
  app = web.application(urls, globals())
  func = app.wsgifunc()

  func = OpenTelemetryMiddleware(func)

  server = wsgi.WSGIServer("/tmp/lr.sock", func, server_name="localhost")
  server.start()
```

Test client
```
 echo -ne 'GET / HTTP/1.1\r\nHost: test.com\r\n\r\n' | nc -U /tmp/lr.sock
HTTP/1.1 200 OK
Transfer-Encoding: chunked
Date: Mon, 07 Jun 2021 16:53:30 GMT
Server: localhost

d
Hello, world!
0
```
Response from the server:

```
Overriding of current TracerProvider is not allowed
{
    "name": "HTTP GET",
    "context": {
        "trace_id": "0xb53b71d4e09f3e00220cbc163f5fadbc",
        "span_id": "0xcb1d28831d509698",
        "trace_state": "[]"
    },
    "kind": "SpanKind.SERVER",
    "parent_id": null,
    "start_time": "2021-06-07T16:53:30.603608Z",
    "end_time": "2021-06-07T16:53:30.615360Z",
    "status": {
        "status_code": "UNSET"
    },
    "attributes": {
        "http.method": "GET",
        "http.server_name": "localhost",
        "http.scheme": "http",
        "http.host": "test.com",
        "http.target": "/",
        "net.peer.port": "",
        "http.flavor": "1.1",
        "http.status_code": 200
    },
    "events": [],
    "links": [],
    "resource": {
        "telemetry.sdk.language": "python",
        "telemetry.sdk.name": "opentelemetry",
        "telemetry.sdk.version": "1.0.0",
        "service.name": "unknown_service"
    }
}
{
    "name": "HTTP GET",
    "context": {
        "trace_id": "0xb53b71d4e09f3e00220cbc163f5fadbc",
        "span_id": "0xcb1d28831d509698",
        "trace_state": "[]"
    },
    "kind": "SpanKind.SERVER",
    "parent_id": null,
    "start_time": "2021-06-07T16:53:30.603608Z",
    "end_time": "2021-06-07T16:53:30.615360Z",
    "status": {
        "status_code": "UNSET"
    },
    "attributes": {
        "http.method": "GET",
        "http.server_name": "localhost",
        "http.scheme": "http",
        "http.host": "test.com",
        "http.target": "/",
        "net.peer.port": "",
        "http.flavor": "1.1",
        "http.status_code": 200
    },
    "events": [],
    "links": [],
    "resource": {
        "telemetry.sdk.language": "python",
        "telemetry.sdk.name": "opentelemetry",
        "telemetry.sdk.version": "1.0.0",
        "service.name": "unknown_service"
    }
}
```

# Does This PR Require a Core Repo Change?
- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [X] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def collect_request_attributes(environ):
     }
 
     host_port = environ.get("SERVER_PORT")
-    if host_port is not None:
+    if host_port is not None and not host_port == '':
```
Comment (codeboten) [inline instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py:121]: Could this be replaced by a check for `host_port`

`if host_port:`

Comment (kamalhussain) [reply instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py:121]: That will not work if host_port has a values like 0, 00, 000 etc

**Review by lzchen — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py (hunk 1 above)
Comment (codeboten) [inline instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py:121]:
```suggestion
    if host_port is not None and not host_port == "":
```

**Review by codeboten — COMMENTED** [review]
Thanks for the fix @kamalhussain, please take a look at my suggestion, I believe it's the only thing needed to get this PR to pass the lint check.

**Pull request merged**